common/metering.go

```diff
@@ -52,28 +60,6 @@ type Gauge interface {
 	ComputationGauge
 }
 
-// CombinedGauge is a Gauge, i.e. it allows metering both memory and computation,
-// by delegating to an independent memory gauge and independent computation gauge.
-//
-// It is mostly just a convenience/helper type, which is useful for e.g. tests,
-// where one has an independent (test) memory gauge, and an independent (test) computation gauge.
-type CombinedGauge struct {
```

Comment:
Remove `CombinedGauge`, as there are two issues:
- It is incomplete: Users may assume that the passed in gauges may be `nil`, but the current implementation does not allow this. Both `MeterMemory` and `MeterComputation` need to be implemented, and they need to check for `nil`
- Such a correct implementation adds another indirection, i.e. overhead